03week/ticTacToe.js

```diff
@@ -24,39 +24,69 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
+  return (board[0][0]=='X' && board[0][1]=='X' && board[0][2]=='X' ||  board[0][0]=='O' && board[0][1]=='O' && board[0][2]=='O') 
+  || (board[1][0]=='X' && board[1][1]=='X' && board[1][2]=='X' || board[1][0]=='O' && board[1][1]=='O' && board[1][2]=='O') 
+  || (board[2][0]=='X' && board[2][1]=='X' && board[2][2]=='X' || board[2][0]=='O' && board[2][1]=='O' && board[2][2]=='O')
 }
 
 function verticalWin() {
-  // Your code here
+  return (board[0][0]=='X' && board[1][0]=='X' && board[2][0]=='X' || board[0][0]=='O' && board[1][0]=='O' && board[2][0]=='O') 
+  || (board[0][1]=='X' && board[1][1]=='X' && board[2][1]=='X' || board[0][1]=='O' && board[1][1]=='O' && board[2][1]=='O') 
+  || (board[0][2]=='X' && board[1][2]=='X' && board[2][2]=='X' || board[0][2]=='O' && board[1][2]=='O' && board[2][2]=='O')
 }
 
 function diagonalWin() {
-  // Your code here
+  return (board[0][0]=='X' && board[1][1]=='X' && board[2][2]=='X' || board[0][0]=='O' && board[1][1]=='O' && board[2][2]=='O') 
+  || (board[2][0]=='X' && board[1][1]=='X' && board[0][2]=='X' || board[2][0]=='O' && board[1][1]=='O' && board[0][2]=='O')
 }
 
 function checkForWin() {
-  // Your code here
+  return horizontalWin() || verticalWin() || diagonalWin()
 }
 
 function ticTacToe(row, column) {
-  // Your code here
+  if(playerTurn=='X'){
+    board[row][column]='X';
+  }
+  else{
+    board[row][column]='O';
+  }
+  if(checkForWin()){
+    console.log("Player " + playerTurn + " won");
+  }
+  else{
+    if(playerTurn=='X'){
+      playerTurn='O'
+    }
+    else{
+      playerTurn='X';
+    }
+  }
+  
 }
 
 function getPrompt() {
   printBoard();
+  if(checkForWin())
```

Comment:
checkForWin should be called in the ticTacToe function, nothing should ever be added to the prompt